addons/account/static/src/js/account_followup_report_widgets.js

```diff
@@ -0,0 +1,165 @@
+            $('#nextActionModal').modal('hide');
+            $('div.page.' + target_id).find('.oe-account-next-action').html(openerp.qweb.render("nextActionDate", {'note': note, 'date': date}));
+        });
+    },
+    setNextAction: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var target_id = $(e.target).parents("div.page").data('context');
+        var dt = new Date();
+        switch($(e.target).data('time')) {
+            case 'one-week':
+                dt.setDate(dt.getDate() + 7);
+                break;
+            case 'two-weeks':
+                dt.setDate(dt.getDate() + 14);
+                break;
+            case 'one-month':
+                dt.setMonth(dt.getMonth() + 1);
+                break;
+            case 'two-months':
+                dt.setMonth(dt.getMonth() + 2);
+                break;
+        }
+        $('.oe-account-picker-next-action-date').data("DateTimePicker").setValue(moment(dt));
+        $("#nextActionModal #target_id").val(target_id);
+        $('#nextActionModal').on('hidden.bs.modal', function (e) {
+            $(this).find('form')[0].reset();
+        });
+        $('#nextActionModal').modal('show');
+    },
+    onChangeBlocked: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var checkbox = $(e.target).is(":checked")
+        var target_id = $(e.target).parents('tr').data('id');
+        var model = new openerp.Model('account.move.line');
+        model.call('write', [[parseInt(target_id)], {'blocked': checkbox}])
+    },
+    onKeyPress: function(e) {
+        var report_name = $("div.page").attr("data-report-name");
+        if ((e.which === 13 || e.which === 10) && (e.ctrlKey || e.metaKey) && report_name == 'followup_report') {
+            $('a.btn-primary.followup-email').trigger('click');
+            var letter_context_list = [];
+            $('a.btn-primary.followup-letter').each(function() {
+                letter_context_list.push($(this).attr('context'))
+            });
+            window.open('?pdf&letter_context_list=' + letter_context_list, '_blank');
+            window.open('?partner_done=all', '_self');
+        }
+    },
+    skipPartner: function(e) {
+        var partner_id = $(e.target).attr("partner");
+        var model = new openerp.Model('res.partner');
+        if ($(e.target).is('a.btn-primary.followup-skip')) {
+            return model.call('update_next_action', [[parseInt(partner_id)]]).then(function (result) {
+                window.open('?partner_done=' + partner_id, '_self');
+            });
+        }
+        else {
+            window.open('?partner_skipped=' + partner_id, '_self');
+        }
+    },
+    printFollowupLetter: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var url = $(e.target).attr("target");
+        window.open(url, '_blank');
+        if ($(e.target).is('.btn-primary')) {
```

Comment:
if I understand correctly what you're doing here is that you've got two button next one to each other and one is btn-primary while the other is not and you made them the contrary, the btn-primary becomes btn-default while the other button becomes btn-primary.
If that's what you do, there is a far easier way to do it in only two lines, you don't even need the if condition
$(e.target).toggleClass('btn-primary btn-default')
$(.followup-skip').toggleClass('btn-primary btn-default')